### PR TITLE
QgsGeometryValidator Improves the error message when the Z differs and adds a location point

### PR DESCRIPTION
**Pull request by lbartoletti**
## Description

As stated in #43582, it is surprising not to have the information on the point that is not closed. This PR fixes this problem.

On the other hand, in the case of a 3D geometry we may need the information if the geometry is closed in 2D and if it is the Zs that are different. I added a method for this with tests and I reuse it in the validator.

### CODE REVIEW

File: python/core/auto_generated/geometry/qgscurve.sip.in [hunk 1]
```diff
@@ -60,6 +60,19 @@ Returns the end point of the curve.
     virtual bool isClosed() const /HoldGIL/;
 %Docstring
 Returns ``True`` if the curve is closed.
+
+.. seealso:: :py:func:`is2DClosed`
+%End
+
+    virtual bool is2DClosed() const /HoldGIL/;
```
Comment (nyalldawson) [inline python/core/auto_generated/geometry/qgscurve.sip.in:67]: I'd suggest "isClosed2D" follows existing method names better

File: tests/src/python/test_qgsgeometryvalidator.py [hunk 2]
```diff
@@ -259,9 +259,20 @@ def test_ring_closed(self):
         validator.run()
         self.assertEqual(len(spy), 1)
 
-        self.assertEqual(spy[0][0].where(), QgsPointXY())
+        self.assertEqual(spy[0][0].where(), QgsPointXY(3, 3))
         self.assertEqual(spy[0][0].what(), 'ring 2 not closed')
 
+        # not closed but 2d close
```
Comment (nyalldawson) [inline tests/src/python/test_qgsgeometryvalidator.py:265]:
```suggestion
        # not closed but 2d closed
```

**Review by nyalldawson — COMMENTED** [review]
Good fix :+1:

**Comment by lbartoletti** [comment]
Thanks! :)

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**